drivers/adc/ad7616/ad7616.c

```diff
@@ -38,20 +40,280 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 *******************************************************************************/
 
+/* Datasheet: https://www.analog.com/media/en/technical-documentation/data-sheets/AD7616.pdf */
+
 /******************************************************************************/
 /***************************** Include Files **********************************/
 /******************************************************************************/
 #include <stdint.h>
-#include <stdio.h>
 #include <stdlib.h>
-#include "spi_engine.h"
-#include "axi_dmac.h"
-#include "no_os_gpio.h"
+#include <string.h>
+#include <assert.h>
+#include <stdio.h>
+
 #include "ad7616.h"
+#include "no_os_gpio.h"
 #include "no_os_error.h"
 #include "no_os_delay.h"
-#include "no_os_axi_io.h"
 #include "no_os_alloc.h"
+#include "no_os_util.h"
+
+#ifdef PLATFORM_ZYNQMP
+#include "spi_engine.h"
+#include "axi_dmac.h"
+#include "no_os_axi_io.h"
+#else
+#include "no_os_spi.h"
+#endif
+
+#define GET_BIT(v, b) ((v >> b) & 1)
+#define TOGGLE_TIMEOUT_DELAY (8192 * 1024)
```

Comment:
I would name these using the `AD7616_` prefix, since there might be a similar `GET_BIT` macro defined in the platform SDKs that we're using.

Reply:
Good idea, fixed